queue/src/main/java/com/bazaarvoice/emodb/queue/core/kafka/KafkaProducerService.java

```diff
@@ -0,0 +1,73 @@
+package com.bazaarvoice.emodb.queue.core.kafka;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.concurrent.Future;
+
+public class KafkaProducerService {
+    private static final Logger _log = LoggerFactory.getLogger(KafkaProducerService.class);
+    private final KafkaProducer<String, String> producer; // Changed to String
+
+    public KafkaProducerService() {
+        this.producer = new KafkaProducer<>(KafkaConfig.getProducerProps());
+        _log.info("KafkaProducerService initialized with producer properties: {}", KafkaConfig.getProducerProps());
+    }
+
+    /**
+     * Sends each message from the collection to the specified Kafka topic separately.
+     *
+     * @param topic   The Kafka topic.
+     * @param events  The collection of messages to be sent.
+     */
+    public void sendMessages(String topic, Collection<String> events, String queueType) {
+        _log.info("Sending {} messages to topic '{}'", events.size(), topic);
+        for (String event : events) {
+            _log.debug("Sending message: {}", event);
+            sendMessage(topic, event,queueType);
+        }
+        _log.info("Finished sending messages to topic '{}'", topic);
+    }
+
+    /**
+     * Sends a single message to the specified Kafka topic.
+     *
+     * @param topic   The Kafka topic.
+     * @param message The message to be sent.
+     */
+    public void sendMessage(String topic, String message, String queueType) {
+        ProducerRecord<String, String> record = new ProducerRecord<>(topic, message);
+        _log.debug("Preparing to send message to topic '{}' with value: {}", topic, message);
+
+        try {
+            Future<RecordMetadata> future = producer.send(record, (metadata, exception) -> {
+                if (exception != null) {
+                    _log.error("Failed to send message to topic '{}'. Error: {}", topic, exception.getMessage());
+                } else {
+                    _log.debug("Message sent to topic '{}' partition {} at offset {}",
+                            metadata.topic(), metadata.partition(), metadata.offset());
+                }
+            });
+            // Optionally, you can wait for the send to complete
+            RecordMetadata metadata = future.get(); // Blocking call
+            _log.info("Message sent successfully to topic '{}' partition {} at offset {}",
+                    metadata.topic(), metadata.partition(), metadata.offset());
+        } catch (Exception e) {
+            _log.error("Failed to send message to topic '{}'. Exception: {}", topic, e.getMessage());
```

Comment:
**issue(blocking):** So we are swallowing the exception here? Meaning that if we fail after a few messages, we keep going with the next ones and may end-up having sent only a partial number of messages but can't recover from this...this will lead to lost messages right? Since the outer `sendMessages` doesn't catch anything either, what is the plan here in terms of truly rethrowing exception and ensuring that messages are re-tried up the stack? is this an early happy path only version of the code?

Reply:
Now rethrowing the exception after logging them, this will be same behaviour of the current system, where end consumers of `send` method will retry untill write to emodb is successful.

Reply:
have you committed these changes...I don't see them?